pytools/graph.py

```diff
@@ -212,10 +212,10 @@ class CycleError(Exception):
     """
     Raised when a topological ordering cannot be computed due to a cycle.
 
-    :attr node: Node in a directed graph that is part of a cycle.
+    :attr nodes: List of nodes in a directed graph that are part of a cycle.
     """
-    def __init__(self, node: T) -> None:
-        self.node = node
+    def __init__(self, nodes: List[T]) -> None:
+        self.nodes = nodes
```

Comment:
This breaks backward compatibility.

Reply:
I think a better approach would be to make `node` a `@property` that spews a `DeprecationWarning`.

Reply:
Implemented in [8823bca](https://github.com/inducer/pytools/pull/159/commits/8823bcad13949b3a14f88c0fb495c9bd494e1101)